src/snowcli/cli/nativeapp/manager.py

```diff
@@ -35,3 +60,101 @@ def deploy_root(self) -> Path:
 
     def build_bundle(self) -> None:
         build_bundle(self.project_root, self.deploy_root, self.artifacts)
+
+    def execute_query_with_role(
+        self, query: str, role: Optional[str] = None
+    ) -> SnowflakeCursor:
+        use_role = f"use role {role}; " if role else ""
+        query = query if query.endswith(";") else f"{query};"
+        try:
+            *_, role_cursor, query_cursor = self._execute_queries(use_role + query)
+            return query_cursor
+        except Exception as err:
+            # Could be ProgrammingError or IntegrityError
+            log.error(err)
+            raise SnowflakeSQLExecutionError(use_role + query)
+
+    def sync_deploy_root_with_stage(self, role: str, app_pkg: str):
+        schema_and_stage: str = self.definition["source_stage"]
+        deploy_root: FilePath = self.definition["deploy_root"]
+
+        # Does a stage already exist within the app pkg, or we need to create one?
+        # Using "if not exists" should take care of either case.
+        log.info("Checking if stage exists, or creating a new one if none exists.")
+        create_stage_if_not_exists_stmt = f"""
+            create stage if not exists {app_pkg}.{schema_and_stage}
+                encryption = (TYPE = 'SNOWFLAKE_SSE');"""
+        self.execute_query_with_role(create_stage_if_not_exists_stmt, role)
+
+        # Perform a diff operation and display results to the user for informational purposes
+        log.info(
+            f"Performing a diff between the Snowflake stage and your local deploy_root {deploy_root} directory."
+        )
+        diff: DiffResult = stage_diff(
+            Path(deploy_root), f"{app_pkg}.{schema_and_stage}"
+        )
+        log.info("Listing results of diff:")
+        print_result(
+            ObjectResult(
+                {
+                    "only on local": ", ".join(diff.only_local),
+                    "only on stage": ", ".join(diff.only_on_stage),
+                    "modified/unknown": ", ".join(diff.different),
+                    "identical": ", ".join(diff.identical),
+                }
+            )
+        )
+
+        # Upload diff-ed files to app pkg stage
+        log.info(
+            f"Uploading diff-ed files from your local {deploy_root} directory to the Snowflake stage."
+        )
+        sync_local_diff_with_stage(
+            role=role, diff_result=diff, stage_path=f"{app_pkg}.{schema_and_stage}"
+        )
+
+    def app_run(self) -> SnowflakeCursor:
+
+        # It is possible that snowflake.yml does not contain some of the optional fields,
+        # and there is no snowflake.local.yml present in the search_path.
+        # In this case, we need to use a generated local yml which provides those default values.
+        generated_local_yml_overrides = generate_local_override_yml(
+            self.definition_manager.project_definition
+        )
```

Comment:
I think overall I'd suggest that we remove `generate_local_override_yml` (as we decided to not automatically generate it, only picking it up if it exists) and instead split out the generated defaults into their own functions. The reason I suggest this is that we don't ever need to generate the file into the yml structure, so it might be easier to just generate individual defaults on-demand when we need them.

Reply:
I've left the original implementation of `generate_local_override_yml` but did away with calling this function.